packages/docusaurus/lib/server/index.js

```diff
@@ -51,18 +51,42 @@ module.exports = async function load(siteDir, cliOptions = {}) {
   const outDir = path.resolve(siteDir, 'build');
   const {baseUrl} = siteConfig;
 
-  // Resolve custom theme override aliases.
-  const themeAliases = await loadTheme(siteDir);
-  // Make a fake plugin to resolve user's theme overrides.
-  if (themeAliases != null) {
-    plugins.push({
-      configureWebpack: () => ({
-        resolve: {
-          alias: themeAliases,
-        },
-      }),
-    });
-  }
+  // Default theme components that are essential and must exist in a Docusaurus app
+  // These can be overriden in plugins/ through component shadowing.
+  // However, we alias it here first as a fallback.
+  const themeFallback = path.resolve(__dirname, '../client/theme-fallback');
```

Comment:
Could we shift these 3 chunks into `loadTheme` and rename existing `loadTheme` to `loadThemeInner` or something? Since we're already doing a lot of splitting this file up into its separate stages, I think we should continue keeping this file clean by shifting the very specific stuff into its own file.